### PR TITLE
Add decision process (alternative proposal)

### PR DESCRIPTION
**Pull request by AndiDog**
This turns out to be pretty much a rewrite of https://github.com/giantswarm/rfc/pull/70, taking out things that we might not need and trying to write down the process as intuitive and minimalistic one. Therefore, please consider this an alternatively-worded replacement of #70.

@JosephSalisbury I decided against merging into the other PR because of the vast changes in text. Think of _this_ PR as alternative suggestion, replacing #70.

Towards https://github.com/giantswarm/giantswarm/issues/24661

### CODE REVIEW

**Review by lyind — COMMENTED** [review]
I like the simplified version better.

However, I can't tell if it works for all stakeholders or if there are any missing rules or details.

File: decision-process/README.md
Comment (JosephSalisbury) [inline decision-process/README.md:6]: Why did you go with URL over just the team name?

File: decision-process/README.md
Comment (JosephSalisbury) [inline decision-process/README.md:39]: When would this be acceptable to be null?

File: decision-process/README.md
Comment (JosephSalisbury) [inline decision-process/README.md:102]: i'd drop this, it feels somewhat like an exception?

File: decision-process/README.md
Comment (JosephSalisbury) [inline decision-process/README.md:40]: i feel like stakeholders is somehow necessary? how can we make sure that relevant parties outside of owners have seen / accepted an RFC?

File: decision-process/README.md
Comment (pipo02mix) [inline decision-process/README.md:79]: We might create this as template file?

File: decision-process/README.md
Comment (pipo02mix) [inline decision-process/README.md:102]: I would mention to elevate in the SIG or last case to Horizon

File: decision-process/README.md
Comment (AndiDog) [reply decision-process/README.md:6]: Same with the `docs` repo (where the field is called `owner` – I changed that to plural _here_). It's also easier to verify whether it points to an actual GitHub group, as we desire here for future automation.

File: decision-process/README.md
Comment (AndiDog) [reply decision-process/README.md:40]: I assume you are asking about getting a minimum number of approvals from relevant teams without forgetting important people – that must be represented as "reviewers" on the PR, since there's no reasonable way to sync `stakeholders: [ ... ]` and PR reviewers (too complex). The chance of forgetting a stakeholder in a YAML field is likely the same as forgetting to set a reviewer on the PR – and adding a team on the PR is just a few clicks and therefore much easier. If you later want to see who reviewed/approved/commented, we just look at the original PR. If there are complaints that a certain team wasn't involved, that's not really what a YAML field can heal, in my opinion, but it could be a mindset or laziness issue, for instance, missing knowledge of the author, or a plain human mistake. That's why I think we should have some very central place where RFCs get announced, and/or regularly discussed. The relevant SIG should take care of adding missing reviewers on the PR. ◀️ Should I add this clarification to the RFC? I agree about the concern and that we should solve it as part of this written process.

If you had concrete ideas about the YAML field, let me know. Maybe you had a good solution in mind?

File: decision-process/README.md
Comment (AndiDog) [reply decision-process/README.md:79]: For personal stuff, I have something like a `create-rfc.sh` script that asks certain information (date, slug, title) and creates the file accordingly. I had a similar idea, but was wondering whether we can make this whole procedure as un-technical as possible so that it's open to anyone. Is there a way to solve this by clicking around GitHub's UI, not requiring a local checkout of the repo, and using the WYSIWYG Markdown editor of GitHub? A template file sounds good if GitHub allows copying that, or has some way to get started writing an RFC without cloning the repo. But maybe I'm mistaken and git is not a hurdle – WDYT?

File: decision-process/README.md
Comment (AndiDog) [reply decision-process/README.md:102]: Ok, will do

File: decision-process/README.md
Comment (AndiDog) [reply decision-process/README.md:131]: Okay, I will add this to the written process

File: decision-process/README.md
Comment (JosephSalisbury) [reply decision-process/README.md:6]: works for me, sure

File: decision-process/README.md
Comment (JosephSalisbury) [reply decision-process/README.md:39]: i think i'm okay with this - my thinking is that i want all (important) rfcs to have some tracking information, but maybe that's too heavyweight

let's go with this for now

File: decision-process/README.md
Comment (JosephSalisbury) [reply decision-process/README.md:40]: my thinking was having a field similar to owners, and then do some CI magic to ensure that at least one person out of the listed teams has approved

my hunch is that having it in the text means that people will consider it more, than if it's in github, but i'm not too sure on that

happy to go either way, i think

**Comment by JosephSalisbury** [comment]
i'll approve once we've got the last comments sorted out, this is a good direction

File: decision-process/README.md
Comment (AndiDog) [reply decision-process/README.md:79]: I'm leaving the template here in the "RFC of RFCs" for now. As long as people use a correct header and h1 heading, it doesn't matter if the file got copy-pasted, came from a template or whatever. People will shout if they think the process is complicated or needs a simple script to create an RFC. We can improve on that when we see it necessary.

File: decision-process/README.md
Comment (AndiDog) [reply decision-process/README.md:102]: Replace the existing bullet point with your suggestion (@JosephSalisbury please comment, since you recommended removing it)

File: decision-process/README.md
Comment (AndiDog) [reply decision-process/README.md:131]: See "Author merges the PR" bullet point where this was already mentioned. I added `#news-*` in the sentence.

File: decision-process/README.md
Comment (AndiDog) [reply decision-process/README.md:40]: I'd recommend we leave this out for now, and if it doesn't work, we can improve the flow by adding the required field

File: decision-process/README.md
Comment (AndiDog) [inline decision-process/README.md:96]: @anvddriesch commented that "no discussion" may not be feasible. Maybe there's a better wording. And I think this item is optional.

File: decision-process/README.md
Comment (AndiDog) [inline decision-process/README.md:98]: @stone-z commented that the RFC author should primarily drive the effort, in their own interest.

@stone-z / @puja108 reported previously using a "review until" date. We'll have to see how this works out culturally. Please comment if you want such a concept in here for the "hard" cases. We discussed in SIG Architecture that RFCs which are important strategically or for the roadmap will usually progress without problems.

File: decision-process/README.md
Comment (AndiDog) [inline decision-process/README.md:95]: @nprokopic commented that the PR should be added to the GitHub team/SIG board

**Review by AndiDog — COMMENTED** [review]
Added commentary from SIG Architecture meeting

File: decision-process/README.md
Comment (nprokopic) [reply decision-process/README.md:40]: My first gut feeling was like Joe's, that we should also have stakeholders, but then I thought that I prefer this RFC being as is right now, simpler.

Then after we we see how this simpler version is working in practice, we can have another RFC which makes stakeholders mandatory, so we can also have e.g. like Joe said CI checks if there is an approval from every stakeholder.

File: decision-process/README.md
Comment (AndiDog) [reply decision-process/README.md:97]: Added as action item

File: decision-process/README.md
Comment (AndiDog) [reply decision-process/README.md]: Reworded

File: decision-process/README.md
Comment (AndiDog) [reply decision-process/README.md]: Mentioned author as primary driver

**Review by JosephSalisbury — APPROVED** [review]
yeah, let's go - @AndiDog can we get some ci magic to check headers? what are next steps for you?

**Comment by AndiDog** [comment]
> yeah, let's go - @AndiDog can we get some ci magic to check headers? what are next steps for you?

I had already prepared rendering of RFCs into a table within the handbook. That helps getting an overview.

And changing the `rfc` repo to check YAML headers shouldn't be that hard.

But let's first get this decision done. There are only 2 approvals by now. When should we go ahead?

**Comment by JosephSalisbury** [comment]
@AndiDog i think we can go ahead now

**Review by nprokopic — APPROVED** [review]
Great improvements to the RFC process 🎉 Thanks @AndiDog and @JosephSalisbury!

**Comment by AndiDog** [comment]
Made the last call in SIG Architecture. Merging this week and going ahead to implementation if there are no more objections or ideas.

**Review by fiunchinho — APPROVED** [review]

**Review by MarcelMue — APPROVED** [review]

**Review by mcharriere — APPROVED** [review]

**Pull request merged**
